services/common/src/redux/slices/spatialDataSlice.ts

```diff
@@ -6,52 +6,90 @@ import { formatDate } from "../utils/helpers";
 import { IGeoJsonFeature } from "@mds/common/interfaces/document/geojsonFeature.interface";
 import { ISpatialBundle } from "@mds/common/interfaces/document/spatialBundle.interface";
 
+interface ItempMineDocumentWithGeomarkId extends IMineDocument {
+  geomark_id?: string;
+}
+
 const createRequestHeader = REQUEST_HEADER.createRequestHeader;
 
 export const spatialDataReducerType = "spatialData";
 
-export const spatialBundlesFromFiles = (files: IMineDocument[]): ISpatialBundle[] => {
-  const geomark_id = "gm-abcdefghijklmnopqrstuvwxyz0000bc";
+export const spatialBundlesFromFiles = async (
+  files: ItempMineDocumentWithGeomarkId[]
+): Promise<ISpatialBundle[]> => {
+  const newFilesWithGeomarkIds = files.some((f) => f.geomark_id);
 
   const temp_indiv = files.filter(
     (f) => f.document_name.endsWith("kmz") || f.document_name.endsWith("kml")
   );
+
+  // Get the core-api spatial bundles.  If these have not yet been created assign the document_name to the mine_document_bundle_id temporarily
   const temp_spatial = files
     .filter((f) => !f.document_name.endsWith("kmz") && !f.document_name.endsWith("kml"))
     .map((f) => {
-      return { ...f, bundle_id: "1" };
+      return {
+        ...f,
+        mine_document_bundle_id: f.mine_document_bundle_id ?? f.document_name.split(".")[0],
+      };
     });
+
   const spatial_bundle_ids = Array.from(
-    new Set(temp_spatial.map((f) => f.bundle_id).filter(Boolean))
+    new Set(temp_spatial.map((f) => f.mine_document_bundle_id).filter(Boolean))
   );
 
-  const spatial_bundles = spatial_bundle_ids.map((id) => {
-    const bundleFiles = temp_spatial
-      .filter((f) => f.bundle_id === id)
-      .map((f) => ({ ...f, key: f.document_manager_guid }));
-    const bundleSize = bundleFiles.length;
-    const document_name = bundleFiles[0].document_name.split(".")[0];
-    const create_user = bundleFiles[0].create_user;
-    const upload_date = bundleFiles[0]?.upload_date
-      ? bundleFiles.sort((a, b) => a.upload_date.localeCompare(b.upload_date))[0].upload_date
-      : formatDate(new Date());
-
-    return {
-      document_name,
-      bundleFiles,
-      upload_date,
-      create_user,
-      bundleSize,
-      bundle_id: id,
-      key: id,
-      isParent: true,
-      geomark_id,
-      isSingleFile: false,
-    };
-  });
-  const individualFiles = temp_indiv
-    .filter((f) => !f.bundle_id)
-    .map((f) => {
+  const headers = createRequestHeader();
+
+  const spatial_bundles = await Promise.all(
```

Comment:
This was a little helper function before, but I think now it wants to be split up into a `fetch` function and a selector to separate concerns and follow the redux pattern.

Reply:
This was the main issue with what took me so long.  I was trying to avoid refactoring all of your previous work since this function is providing the base for two seperate components.  One of those components just needs the file grouping and the other needs the geomark_id which sometimes needs to be fetched.

Since the state of the files (whether or not the mine_document_bundle has been created) is what dictates where it looks for the geomark_id it starts to get really janky to split up the concerns. 

I'll take another look at it.

Reply:
I do appreciate the eagerness, but this PR is not yet "Ready For Review"